File: 1200. Minimum Absolute Difference/1200.py
class Solution:
    def minimumAbsDifference(self, arr: list[int]) -> list[list[int]]:
        arr.sort()
        answer = []
        min_sum = float('inf')

        for i in range(len(arr)-1):
            diff = arr[i+1] - arr[i]

            if diff < min_sum:
                min_sum = diff
                answer = [[arr[i], arr[i+1]]]

            elif diff == min_sum:
                answer.append([arr[i], arr[i+1]])


        return answer



S = Solution()
print(S.minimumAbsDifference([3,8,-10,23,19,-4,-14,27]))

File: 1200. Minimum Absolute Difference/1200.cs
public class Solution {
    public IList<IList<int>> MinimumAbsDifference(int[] arr) {
        Array.Sort(arr);

        List<IList<int>> answer = new List<IList<int>>();
        int minSum = int.MaxValue;

        for (int i = 0; i < arr.Length -1 ; i++)
        {
            int diff = arr[i+1] - arr[i];

            if (diff < minSum)
            {
                minSum = diff;
                answer.Clear();
                answer.Add(new List<int> { arr[i], arr[i+1] });
            } else if (diff == minSum)
            {
                answer.Add(new List<int> { arr[i], arr[i+1] });
            }
        }
        return answer;
        
    }
}